### PR TITLE
TFAC-698 Exclude Vanity URLs from Indexing

### PR DESCRIPTION
**Pull request by jedtan**

### CODE REVIEW

File: gatsby-config.esm.js [hunk 1]
```diff
@@ -61,6 +61,33 @@ module.exports = {
...
+              context {
+                referrer {
+                  id
+                }
+              }
+            }
+          }
+        }`,
+        resolvePages: ({ allSitePage: {nodes: allPages}}) => allPages,
+        filterPages: (page, excludedRoute) => page.context.referrer !== null || page.path.slice(0, -1) === excludedRoute,
```
Comment (jedtan) [inline gatsby-config.esm.js:83]: The slice is for removing trailing slashes in the page.path.

Comment (kmjennison) [reply gatsby-config.esm.js:83]: Please add this as a code comment.

File: src/components/V4HomePage.js [hunk 2]
```diff
@@ -86,6 +88,7 @@ const HomepageWrapper = ({
           />
           <Helmet>
             <link rel="canonical" href={canonicalURL} />
+            {referrer ? <meta name="robots" content="noindex" /> : null}
```
Comment (kmjennison) [inline src/components/V4HomePage.js:91]: Please add tests for this.

File: src/pages/index.js [hunk 3]
```diff
@@ -228,6 +228,9 @@ class IndexPage extends React.Component {
         <div>
           <Helmet>
             <link rel="canonical" href={canonicalURL} />
+            {this.isReferralFromChannel() ? (
+              <meta name="robots" content="noindex" />
+            ) : null}
```
Comment (kmjennison) [inline src/pages/index.js:233]: Please add tests for this.

File: src/pages/cats.js [hunk 4]
```diff
@@ -199,6 +200,7 @@ const Cats = ({ pageContext, location }) => {
         STORAGE_REFERRAL_DATA_REFERRING_CHANNEL,
         referrerId
       )
+      setIsReferralChannel(true)
   }, [])
```
Comment (kmjennison) [inline src/pages/cats.js:204]: Fix lint failures.

**Review by kmjennison — CHANGES_REQUESTED** [review]

File: src/components/__tests__/TeamSeas.Install.test.js [hunk 5]
```diff
@@ -1,11 +0,0 @@
-/* eslint-env jest */
```
Comment (jedtan) [inline src/components/__tests__/TeamSeas.Install.test.js:1]: This file had just one test which is duplicated in the other TeamSeasInstall component test file, so I just deleted this.

**Review by kmjennison — APPROVED** [review]
LGTM! 🦊

**Pull request merged**